### ISSUE TITLE
Merge function should overwrite arrays

### ISSUE THREAD

**Issue by rohit-smpx** [smartprix/cfg#5]
```js
const cfg = require('@smpx/cfg');

// Init key
cfg.merge({testKey: ['a', 'b']});
console.log(cfg('testKey')); // [ 'a', 'b' ]

cfg.merge({testKey: ['c']});
console.log(cfg('testKey')); // ['c', 'b']
```

While the expected value for the test key is `['c']`

**Comment by shahbaz-smpx** [comment]
This is due to the fact that we're recursively iterating key of an object, in case of arrays, if my new array value for an existing key is smaller than the previous value, then it would just modify the value up to that point. E.g. in your example: length of the original value for key `testKey` is 2, while that of new value is 1, so this method just modify the array up to position 1 (0th in array terms), other position in array remains unchanged.

What can be done for that? We can probably add a check of length in case of arrays and pop the unhanged elements by looping through the modified array.
```js
    if (oldVal instanceof Array && oldVal.length > newVal.length) {
        for(i =newVal.length; i < oldVal.length; i++) {
	    oldVal.pop(i);
        }
    }
```

or we could pre-check the value for Array (before iterating over keys) deep copy the newVal to oldVal and return the newVal object

```js
    if (oldVal instanceof Array) {
        oldVal = [...newVal]
	return obj;
    }
```
I guess there may be better ways. Please suggest if any, or should I continue with any one of them?

**Comment by hit12369** [comment]
This is because the `isObject` function does not correctly check for plain objects. It'll return true for all kind of objects and classes.

Correct way to check for a plain object would be:
```js
Object.prototype.toString.call(obj) === '[object Object]'
```

**Comment by hit12369** [comment]
Fixed in `1.2.2`
